syntax = "proto3";

package infrastructure.network.grpc;

service GitRepoFoundService {
  // Receive a git_repo_found event
  rpc GitRepoFoundNotifications (GitRepoFound) returns (Reply) {}
}

// The git_repo_found notification
message GitRepoFound {
  string package_name = 1;
  string package_version = 2;
  string url = 3;
  string tag = 4;
  map<string, string> metadata = 5;
  string subfolder = 6;
}

// The response message containing a status code and
message Reply {
  int32 code = 1;
  string message = 2;
}
